pkg/shardddl/optimism/lock.go

```diff
@@ -72,7 +72,7 @@ func NewLock(ID, task, downSchema, downTable string, ti *model.TableInfo, tts []
 		synced:     true,
 		versions:   make(map[string]map[string]map[string]int64),
 	}
-	l.addTables(tts)
+	l.addTables(l.joined, tts)
```

Comment:
What if `tts` contains the other tables whose schema didn't reach `l.joined`?

Reply:
Here l.joined equals tableInfoBefore. If it's a new lock(first table), tableInfoBefore=l.joined. For other newer tables, they will be added in L163

Reply:
The `tts` here is from https://github.com/pingcap/dm/blob/master/dm/master/shardddl/optimist.go#L267. `optimistic` will get all source tables from etcd and this `tts` may have not only the table in this info, I think.

Reply:
em... That's right, I will fix it